lib/livebook/teams/requests.ex

```diff
@@ -305,6 +343,14 @@ defmodule Livebook.Teams.Requests do
 
   defp transform_response({request, response}) do
     case {request, response} do
+      {request, %{status: 404}} when request.private.cli and request.private.deploy ->
```

Comment:
In this case, since we want to identify exactly why it happened, let's change the server to return a 404 with JSON or a 422, so we don't have to guess from here.